static/app/types/onboarding.tsx

```diff
@@ -49,7 +58,26 @@ export type OnboardingTaskDescriptor = {
       action: () => void;
       actionType: 'action';
     }
-);
+) &
+  (
+    | {
+        Component: (OnboardindgCustomComponentProps) => JSX.Element | null;
+        /**
+         * If true, the `SupplementComponent` will be used to render the entire card,
+         * and the card will be rendered before any other cards regardless of completion status.
+         * `SupplementComponent` is therefore responsible for determining the completion status
+         * of the card by returning null when it's completed.
+         */
+        customized: true;
```

Comment:
@Neo-Zhixing why do we need this extra prop? Can't we just look at if `Component` exists as a prop?

Reply:
Ok I guess that could work too. Good call.